components/coreapi/blocks.go

```diff
@@ -28,13 +29,18 @@ func blockMetadataResponseByID(c echo.Context) (*blockMetadataResponse, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// TODO: fill in blockReason, TxState, TxReason.
 	bmResponse := &blockMetadataResponse{
-		BlockID:    block.ID().ToHex(),
-		BlockState: blockStatePending.String(),
+		BlockID:            block.ID().ToHex(),
+		StrongParents:      block.Block().StrongParents.ToHex(),
+		WeakParents:        block.Block().WeakParents.ToHex(),
+		ShallowLikeParents: block.Block().ShallowLikeParents.ToHex(),
+		BlockState:         blockStatePending.String(),
 	}
 	_, exists := deps.Protocol.MainEngineInstance().Block(block.ID())
 	if !exists {
-		bmResponse.BlockError = "block not found"
+		bmResponse.BlockStateReason = "block not found"
```

Comment:
where does this reason as a free-text come from? If we are using text here these need to be defined in the API so that every implementation matches

Reply:
We don't even need the codes from line 41, if there's any issue fetching a block, it should return an error in the upper section.